src/Input/AbstractParamAwareInput.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+/**
+ * @see       https://github.com/laminas/laminas-cli for the canonical source repository
+ * @copyright https://github.com/laminas/laminas-cli/blob/master/COPYRIGHT.md
+ * @license   https://github.com/laminas/laminas-cli/blob/master/LICENSE.md New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Laminas\Cli\Input;
+
+use InvalidArgumentException;
+use Symfony\Component\Console\Helper\QuestionHelper;
+use Symfony\Component\Console\Input\InputDefinition;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\StreamableInputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\Question;
+
+use function is_array;
+use function sprintf;
+
+/**
+ * Decorate an input instance to add a `getParam()` method.
+ *
+ * @internal
+ */
+abstract class AbstractParamAwareInput implements ParamAwareInputInterface
+{
+    /** @var QuestionHelper */
+    protected $helper;
+
+    /** @var InputInterface */
+    protected $input;
+
+    /** @var OutputInterface */
+    protected $output;
+
+    /** @var array array<string, InputParamInterface> */
+    protected $params;
+
+    /**
+     * @param array $params array<string, InputParamInterface>
+     */
+    public function __construct(InputInterface $input, OutputInterface $output, QuestionHelper $helper, array $params)
+    {
+        $this->input  = $input;
+        $this->output = $output;
+        $this->helper = $helper;
+        $this->params = $params;
+    }
+
+    /**
+     * Define this method in order to modify the question, if needed, before
+     * prompting for an answer.
+     */
+    abstract protected function modifyQuestion(Question $question): void;
+
+    /**
+     * @return mixed
+     * @throws InvalidArgumentException When the parameter does not exist.
+     * @throws InvalidArgumentException When the parameter is of an invalid type.
+     * @throws InvalidArgumentException When the parameter is required, input is
+     *     non-interactive, and no value is provided.
+     */
+    public function getParam(string $name)
+    {
+        if (! is_array($this->params) || ! isset($this->params[$name])) {
```

Comment:
`! isset($this->params[$name])` only check should be enough as next in line 74 is try get `$inputParam = $this->params[$name];`

Reply:
also, the `params` property itself already used `array` in constructor parameter.